daemon/cmd/daemon_main.go

```diff
@@ -1801,12 +1802,13 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 			// where policy synchronization is not done via k8s. Also in k8s
 			// case it is prudent to allow concurrent endpoint regenerations to
 			// (re-)allocate the restored identities before we release them.
-			time.Sleep(option.Config.IdentityRestoreGracePeriod)
 			log.Debugf("Releasing reference counts for %d restored CIDR identities", len(d.restoredCIDRs))
 			d.ipcache.ReleaseCIDRIdentitiesByCIDR(d.restoredCIDRs)
 			// release the memory held by restored CIDRs
 			d.restoredCIDRs = nil
 		}
+		// Release restored node CIDR identities as well.
+		d.nodeDiscovery.Manager.ReleaseRestoredIdentities()
```

Comment:
Shouldn't we have similar logic to above releasing of CIDRs? As in, check that there are any restored identities and then release them after grace period instead of always calling `Sleep`?

Reply:
We do; that's what this function does. (It is going to go away based on Joe's feedback, however).